### PR TITLE
Output binlog position, make xid/commit configurable

### PR DESCRIPTION
**Pull request by dadah89**
/cc @zendesk/rules

### Description

Example:

>23:05:20,149 DEBUG MaxwellKafkaProducer -    {"database":"zendesk_development_shard_1","table":"tickets","type":"update","ts":1472684720,"xid":305865,"commit":true,"position":"master.000358:29171019","data":{"id":10001,"account_id":1,"requester_id":10006,"submitter_id":10001,"assignee_id":10001,"group_id":10001,"status_id":1,"priority_id":2,"via_id":51,"ticket_type_id":2,"created_at":"2016-08-25 22:53:13","updated_at":"2016-08-31 23:05:20","description":"Hi Admin,\n\nEmails, chats, voicemails, and tweets are captured in Zendesk as tickets. Start typing above to respond and click Submit to send. To test how an email becomes a ticket, send a message to support@support.zendesk.com.\n\nCurious about what your customers will see when you reply? Check out this video:\nhttps://demos.zendesk.com/hc/en-us/articles/202341799\n","assignee_updated_at":"2016-08-25 22:53:14","requester_updated_at":"2016-08-25 22:53:13","assigned_at":"2016-08-25 22:53:13","status_updated_at":"2016-08-25 22:53:13","nice_id":1,"initially_assigned_at":"2016-08-25 22:53:13","current_tags":"sample zendesk","updated_by_type_id":0,"subject":"Sample ticket: Meet the ticket","base_score":12,"generated_timestamp":"2016-08-31 23:05:20","satisfaction_score":0,"locale_id":1,"latest_comment_added_at":"2016-08-25 22:53:13","latest_public_comment_added_at":"2016-08-25 22:53:13","brand_id":10001,"inbox_updated_at":"2016-08-25 22:53:13","sla_breach_status":4294967295,"satisfaction_reason_code":0,"number_of_incidents":0,"is_public":1},"old":{"updated_at":"2016-08-26 21:41:56","generated_timestamp":"2016-08-26 21:41:56"}}

### Risks

Low.

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 1]
```diff
@@ -76,6 +78,9 @@ protected OptionParser buildOptionParser() {
 
 		parser.accepts( "producer", "producer type: stdout|file|kafka" ).withRequiredArg();
 		parser.accepts( "output_file", "output file for 'file' producer" ).withRequiredArg();
+		parser.accepts( "output_includes_binlog_position", "produced records include binlog position" );
+		parser.accepts( "output_includes_commit", "produced records include commit" );
+		parser.accepts( "output_includes_xid", "produced records include xid" );
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: i think just combine these two fields into one flag `commit_info`

Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: leave out the word `includes` -- so `output_binlog_position`, `output_commit_info`

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 2]
```diff
@@ -246,6 +251,11 @@ private void setup(OptionSet options, Properties properties) {
 		if ( options != null && options.has("replay")) {
 			this.replayMode = true;
 		}
+
+		boolean outputIncludesBinlogPosition = options != null && options.has("output_includes_binlog_position");
+		boolean outputIncludesCommit = options != null && options.has("output_includes_commit");
+		boolean outputIncludesXid = options != null && options.has("output_includes_xid");
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: need to check the `properties` here too

Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: gonna copy https://github.com/zendesk/maxwell/pull/420/files#diff-d24312bad9cc0cd2c507665e889b2c1aR146 so we don't end with two implementations of the same thing

**Comment by dadah89** [comment]
@osheroff updated

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 3]
```diff
@@ -246,6 +260,10 @@ private void setup(OptionSet options, Properties properties) {
 		if ( options != null && options.has("replay")) {
 			this.replayMode = true;
 		}
+
+		boolean outputBinlogPosition = fetchBooleanOption("output_binlog_position", options, properties, false);
+		boolean outputCommitInfo = fetchBooleanOption("output_commit_info", options, properties, false);
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: default this to true

**Comment by osheroff** [comment]
update the documentation.

**Comment by osheroff** [comment]
after that we're good to go.

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 4]
```diff
@@ -76,6 +78,8 @@ protected OptionParser buildOptionParser() {
 
 		parser.accepts( "producer", "producer type: stdout|file|kafka" ).withRequiredArg();
 		parser.accepts( "output_file", "output file for 'file' producer" ).withRequiredArg();
+		parser.accepts( "output_binlog_position", "produced records include binlog position" );
+		parser.accepts( "output_commit_info", "produced records include commit and xid" );
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: move these to their own section below the producer options

**Comment by osheroff** [comment]
local test looks good

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java (hunk 3 above)
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: should I change the flag to accept arguments (true/false?) or rename it `no_output_commit_info`?

Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: oh, I see.  I guess it'll have to take true/false.

**Comment by dadah89** [comment]
@osheroff updated

File: config.properties.example [hunk 5]
```diff
@@ -136,3 +136,9 @@ password=maxwell
 
 # output filename when using the "file" producer
 #output_file=/path/to/file
+
+# produced records include binlog position
+output_binlog_position=true
```
Comment (osheroff) [inline config.properties.example]: leave the actual commands commented

**Comment by osheroff** [comment]
👍

**Pull request merged**